agent/structs/connect_proxy_config.go

```diff
@@ -247,6 +247,10 @@ type Upstream struct {
 
 	// MeshGateway is the configuration for mesh gateway usage of this upstream
 	MeshGateway MeshGatewayConfig `json:",omitempty"`
+
+	// IngressHosts are a list of hosts that should route to this upstream from
+	// an ingress gateway
+	IngressHosts []string `json:"-" bexpr:"-"`
```

Comment:
We added this field on the `structs.Upstream` struct, feels a bit out of place at the minute though. Thoughts on whether this is a good place to add the field?

Reply:
We added a test in agent/structs/connect_proxy_config_test.go that ensures this field does not marshal/unmarshal into JSON.